utils/modular_model_converter.py

```diff
@@ -1101,9 +1113,18 @@ def _recursively_add_all_new_needed_functions_in_files(self):
                             dependency, body, self.all_definitions[dependency], parent=parent
                         )
 
+    def _filter_imports_for_file(self, imports):
+        _dict = defaultdict(dict)
+        for key, value in imports.items():
+            if key in self.function_call_class_mapping:
+                node = self.function_call_class_mapping[key]
+                if len(node) == 1:
+                    file_name = self.class_to_file_type[node.copy().pop()]
+                    _dict[file_name][key] = value
+        return _dict
+
     def leave_Module(self, original_node: cst.Module, node):
-        imports = {self.python_module.code_for_node(k): k for k in self.all_imports}
-        dependency_imports = {file_type: imports.copy() for file_type in self.files}
+        dependency_imports = self._filter_imports_for_file(self.all_imports.copy())
         for super_file_name, visiter in self.visited_module.items():
             file_type = re.search(r"models?\.\w*?\.(\w*?)_", super_file_name).groups()[0]
             dependency_imports[file_type].update(
```

Comment:
Here imports from the dependency files are just dumped as is, without being filtered. In Glm, this causes a duplicated import because one import from Phi3 is protected, and the same import is unprotected in other files.